python/ray/experimental/channel/nccl_group.py

```diff
@@ -0,0 +1,84 @@
+from typing import Dict, Optional
+
+import ray
+from ray.util.collective.collective_group import nccl_util
+
+try:
+    import torch
+except ImportError:
+    torch = None
+
+
+class _NcclGroup:
+    def __init__(
+        self,
+        world_size: int,
+        comm_id: int,
+        rank: Optional[int],
+        actor_ids_to_ranks: Dict[ray.ActorID, int],
+        cuda_stream: Optional[int],
+    ):
+        self._rank: Optional[int] = rank
+        if rank is not None:
+            assert ray.get_gpu_ids(), "NCCL actor has no GPUs assigned"
+
+            from cupy.cuda import nccl
+
+            self._comm = nccl.NcclCommunicator(world_size, comm_id, rank)
+        else:
+            # Driver does not have a rank.
+            self._comm = None
+        self._actor_ids_to_ranks = actor_ids_to_ranks
+
+        self._cuda_stream = cuda_stream
```

Comment:
if it is None, should we create a custom stream? Otherwise, it will use the default stream which can be used by other libs like torch? (or is this intended behavior?)

Reply:
Ah yeah, right now the intended behavior is to serialize with torch. Was planning to allow more flexible stream management stuff in followup.